rust/rbac-registration/src/cardano/cip509/utils/cip134_uri_set.rs

```diff
@@ -90,7 +106,7 @@ impl Cip0134UriSet {
         result
     }
 
-    /// Returns a list of stake addresses by the given role.
+    /// Returns a set of stake addresses by the given role.
     #[must_use]
     pub fn role_stake_addresses(
```

Comment:
As for now, the behavior of `role_stake_addresses` and `role_addresses` functions is inconsistent with the one of the `stake_addresses` function. The latter one doesn't include taken addresses, but the former ones do.

We should probably add unit tests for these methods.